### ISSUE TITLE
Missing API docs!

### ISSUE THREAD

**Issue by niranjan-talkinglands** [geoman-io/maplibre-geoman#5]
There is no docs on the inbuilt APIs or proper docs on draw Events. Please add it asap.

**Comment by instantgis** [comment]
I second that. Granted its "just launched" but that's no excuse for meager docs.
I had to dig deep in all the samples to discover how to even configure maplibre-geoman.
I am using it with Angular 18 and Mapbox.
_Mapbox-gl-draw has not budged in months and is lacking features for my app._
This is THE alternative?

```
  ngOnInit(): void {
    const gmOptions: GmOptionsPartial = {
      controls: {
        draw: {
          marker: { uiEnabled: false, active: false },
          circle: { uiEnabled: false, active: false },
          circle_marker: { uiEnabled: true, active: false },
          text_marker: { uiEnabled: false, active: false },
          line: { uiEnabled: false, active: false },
          rectangle: { uiEnabled: false, active: false },
          polygon: { uiEnabled: false, active: false },
          freehand: { uiEnabled: false, active: false },
          custom_shape: { uiEnabled: false, active: false },
        },
        helper: {
          shape_markers: { uiEnabled: false, active: false },
          pin: { uiEnabled: false, active: false },
          snapping: { uiEnabled: false, active: false },
          snap_guides: { uiEnabled: false, active: false },
          measurements: { uiEnabled: false, active: false },
          auto_trace: { uiEnabled: false, active: false },
          geofencing: { uiEnabled: false, active: false },
          click_to_edit: { uiEnabled: false, active: false },
          zoom_to_features: { uiEnabled: true, active: false },
        },
        edit: {
          drag: { uiEnabled: true, active: false },
          change: { uiEnabled: false, active: false },
          rotate: { uiEnabled: false, active: false },
          scale: { uiEnabled: false, active: false },
          copy: { uiEnabled: false, active: false },
          cut: { uiEnabled: false, active: false },
          split: { uiEnabled: false, active: false },
          union: { uiEnabled: false, active: false },
          difference: { uiEnabled: false, active: false },
          line_simplification: { uiEnabled: false, active: false },
          lasso: { uiEnabled: false, active: false },
          delete: { uiEnabled: true, active: false },
        },
      },
    };

    this.map = new mapboxgl.Map({
      accessToken: environment.mapbox.accessToken,
      container: 'map',
      style: this.style,
      zoom: DEFAULT_ZOOM,
      center: [this.lng, this.lat],
      projection: 'globe',
    });
    this.geoman = new Geoman(this.map, gmOptions);
    this.map.on('load', () => {
      console.log('Map loaded');
    });
    this.map.on('gm:loaded', () => {
      console.log('Geoman fully loaded');
    });
    this.map.on('gm:create', (event: GMEvent) => {
      console.log('Event', event);
    });
    if (this.map.gm) {
      this.map.gm.setGlobalEventsListener(
        (event: GlobalEventsListenerParameters) => {
          if (event.type === 'converted') {
            console.log('Regular event', event);
          } else if (event.type === 'system') {
            console.log('System event', event);
          }
        },
      );
    }
  }

```

Why does it bark at you even though you turn non free features off?

```
Mode "draw:freehand" is not available
Mode "draw:custom_shape" is not available
Mode "edit:scale" is not available
Mode "edit:copy" is not available
Mode "edit:split" is not available
Mode "edit:union" is not available
Mode "edit:difference" is not available
Mode "edit:line_simplification" is not available
Mode "edit:lasso" is not available
Mode "helper:pin" is not available
Mode "helper:snap_guides" is not available
Mode "helper:measurements" is not available
Mode "helper:auto_trace" is not available
Mode "helper:geofencing" is not available
Mode "helper:click_to_edit" is not available
```

Would I fork out 1K euros on it?
Hummm not yet.
It's not ready for prime time yet.

**You can make it a lot better I am sure.**

**Comment by mscno** [comment]
Hi @instantgis thanks a lot for the comment.

We spent a lot of energy in the past weeks on achieving support for mapbox (initially we only supported maplibre), and this took time away from updating docs. The mapbox support is still experimental, but we are continuously on improving it. I will push updated docs later today, with better info on how to configure maplibre-geoman, as well as the public api for drawing.

Regarding the free/pro feature warnings, we will also tone those down, so that they only appear if you enable the modes in free version.
